src/main/java/org/gradle/playframework/plugins/PlayDistributionPlugin.java

```diff
@@ -182,7 +182,8 @@ private void createDistributionZipTasks(Project project, Distribution distributi
             return providers.provider(new Callable<String>() {
                 @Override
                 public String call() throws Exception {
-                    return (distribution.getBaseName() != null && "".equals(distribution.getBaseName())) ? distribution.getBaseName() : distribution.getName();
+                    String baseName = (String) Distribution.class.getMethod("getBaseName").invoke(distribution);
+                    return "".equals(baseName) ? "" : distribution.getName();
```

Comment:
Shouldn't this be hardened against the eventual removal of `getBaseName`. IE. shouldn't this not fail when `getBaseName` isn't a method anymore?

Reply:
This path of the code won't be executed with _Gradle versions < 6.0_. In these cases the method will always be there. And we have test coverage to prove that.